### DISCUSSION TITLE
[Bug] How create Custom Switch in xamarin

### DISCUSSION THREAD

**Discussion by Shraddha996** [xamarin/Xamarin.Forms#11305] [General]
Hello,
As am new to this platform I am unable to find a proper solution for how to create a 
custom switch with label on it

![2](https://user-images.githubusercontent.com/39983941/86488290-26a38100-bd7e-11ea-9032-7fdb93a5a11d.PNG)

I want a switch like this.
Please help!!!!!!!

**Comment by davidortinau** [comment]
Similar to #11304 you can create this using perhaps 2 buttons in a grid, a frame or shape for the selection indicator, and use VisualStateManager to change the styles. If you want animation and wish to stay in XAML, then check out [Xamanimation](https://github.com/jsuarezruiz/Xamanimation). Or you can do this in C# and have a [simple view animation](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/animation/simple) like:

```csharp
if(isChecked){
  selectionIndicator.TranslateTo(100,0,500);
}else{
  selectionIndicator.TranslateTo(0,0,500);
}
```
